barretenberg/cpp/src/barretenberg/crypto/merkle_tree/node_store/cached_content_addressed_tree_store.hpp

```diff
@@ -662,25 +705,24 @@ void ContentAddressedCachedTreeStore<LeafValueType>::commit(TreeMeta& finalMeta,
                 persist_leaf_indices(*tx);
             }
             // If we are commiting a block, we need to persist the root, since the new block "references" this root
-            // However, if the root is the empty root we can't persist it, since it's not a real node
-            // We are abusing the trees in some tests, trying to add empty blocks to initial empty trees
-            // That is not expected behavior since the unwind operation will fail trying to decrease refcount
-            // for the empty root, which doesn't exist.
-            if (dataPresent || (asBlock && uncommittedMeta.size > 0)) {
+            // However, if the root is the empty root we can't persist it, since it's not a real node and doesn't have
+            // nodes beneath it. We coujld store a 'dummy' node to represent it but then we have to work around the
+            // absence of a real tree elsewhere. So, if the tree is completely empty we do not store any node data, the
+            // only issue is this needs to be recognised when we unwind or remove historic blocks i.e. there will be no
+            // node date to remove for these blocks
+            if (dataPresent || uncommittedMeta.size > 0) {
```

Comment:
The actual fix is subtle. If the very first block is empty then there is no data in the tree and the root is the 'zero' root. In this situation we can't write nodes to the store as to do so is meaningless (should we write the entire tree?!). I did wonder if we should write the root node, but this causes lots of problems elsewhere as there is no tree beneath it.

So we continue to leave the node store unpopulated until such time as the tree becomes populated with at least one leaf/node.